src/aiperf/zmq/sub_client.py

```
@@ -139,6 +145,29 @@ async def _subscribe_internal(
+        self, callback: Callable[[Message], Awaitable[None]]
+    ) -> None:
+        """Subscribe to all messages.
+
+        Args:
+            callback: Coroutine to call when a message is received (receives Message object)
+        """
+        try:
+            if self._wildcard_subscriber is not None:
+                raise CommunicationError(
+                    "Wildcard subscriber already set. Only one wildcard subscriber is allowed."
+                )
```

Ruff (0.15.2)

[warning] Abstract `raise` to an inner function

(TRY301)